src/main/java/org/candlepin/subscriptions/db/SubscriptionRepository.java

```diff
@@ -59,4 +60,15 @@
         "o.role IN :#{#roles})")
     List<Subscription> findSubscriptionByAccountAndUsageKey(@Param("accountNumber") String accountNumber,
         @Param("key") UsageCalculation.Key usageKey, @Param("roles") Set<String> roles);
+
+    @Query("SELECT s FROM Subscription s WHERE s.accountNumber = :accountNumber AND " +
+        "s.sku = ALL (SELECT DISTINCT o.sku FROM Offering o WHERE " +
+        ":#{#key.usage} = o.usage AND " +
+        ":#{#key.sla} = o.serviceLevel AND " +
+        "o.role IN :#{#roles}) AND s.startDate > :rangeStart AND s.endDate < :rangeEnd AND " +
```

Comment:
```suggestion
        "o.role IN :#{#roles}) AND s.startDate <= :rangeStart AND s.endDate >= :rangeEnd AND " +
```
Subscription needs to start before the usage can be reported against, and it must end after the usage period end.